src/main/kotlin/tech/thatgravyboat/skyblockpv/screens/tabs/ChocolateFactoryScreen.kt

```diff
@@ -0,0 +1,313 @@
+                    italic = false
+                    color = TextColor.GRAY
+                    append("Chocolate Production ") {
+                        color = TextColor.GOLD
+                    }
+                }
+                add("for ") {
+                    italic = false
+                    color = TextColor.GRAY
+                    append("1h ") {
+                        color = TextColor.GREEN
+                    }
+                    append("per charge.")
+                }
+                space()
+                add("Charges: ") {
+                    italic = false
+                    color = TextColor.GRAY
+                    append("${cf.timeTower?.charges ?: 0}") {
+                        color = TextColor.LIGHT_PURPLE
+                    }
+                    append("/") {
+                        color = TextColor.GRAY
+                    }
+                    append("3") {
+                        color = TextColor.LIGHT_PURPLE
+                    }
+                }
+            }
+            createUpgradeItem(Items.RABBIT_FOOT, "Rabbit Shrine", cf.rabbitRarityUpgrades) {
+                add("Increases the chance of getting ") {
+                    italic = false
+                    color = TextColor.GRAY
+                }
+                add("higher rarity rabbits ") {
+                    italic = false
+                    color = TextColor.LIGHT_PURPLE
+                    append("during ") {
+                        color = TextColor.GRAY
+                    }
+                    append("Hoppity's Hunt") {
+                        color = TextColor.LIGHT_PURPLE
+                    }
+                    append(".") {
+                        color = TextColor.GRAY
+                    }
+                }
+            }
+            createUpgradeItem(coachSkull.copy(), "Coach Jackrabbit", cf.chocolateMultiplierUpgrades) {
+                add("Increases the amount of ") {
+                    italic = false
+                    color = TextColor.GRAY
+                }
+                add("Chocolate ") {
+                    italic = false
+                    color = TextColor.GOLD
+                    append("you get per second.") {
+                        color = TextColor.GRAY
+                    }
+                }
+            }
+        }.map { Displays.padding(2, Displays.item(it, showTooltip = true)) }.toRow(2).let {
+            Displays.background(SkyBlockPv.id("inventory/inventory-4x1"), Displays.padding(2, it))
+        }.asWidget(),
+    )
+
+    private fun MutableList<ItemStack>.createUpgradeItem(item: ItemStack, name: String, level: Int, tooltipBuilder: TooltipBuilder.() -> Unit) = add(
+        item.apply {
+            set(DataComponents.CUSTOM_NAME, Text.join(name, " $level") { italic = false; color = TextColor.LIGHT_PURPLE })
+            set(DataComponents.LORE, ItemLore(TooltipBuilder().apply(tooltipBuilder).build().split("\n")))
```

Comment:
pass the list of components as both first and 2nd argument to avoid the automatic purple and italic style